### PR TITLE
Add test for calling CDI.current inside a new thread

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/src/com/ibm/ws/cdi12/test/common/web/CDICurrentTestServlet.java [hunk 1]
```diff
@@ -37,4 +46,65 @@ public void testCDICurrent() {
...
+        while (i > 0) {
+            i--;
+            if (wasCDICurrentFound != null) {
+                assertTrue("CDI.current returned null when called in a new Thread", wasCDICurrentFound);
+                return;
+            }
+            Thread.sleep(5);
+        }
+    }
+    */
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/src/com/ibm/ws/cdi12/test/common/web/CDICurrentTestServlet.java:76]: No need to keep this whole test but a good explaination in a comment would still be useful.

File: dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/resources/META-INF/application.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/resources/META-INF/application.xml:1]: you're right, this shouldn't be needed

File: dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/src/com/ibm/ws/cdi12/test/common/web/CDICurrentTestServlet.java [hunk 3]
```diff
@@ -37,4 +46,65 @@ public void testCDICurrent() {
...
+            if (wasCDICurrentFoundViaMOE != null) {
+                assertTrue("CDI.current returned null when called in a new Thread", wasCDICurrentFoundViaMOE);
+                return;
+            }
+            Thread.sleep(5);
+        }
+    }
+
+
+    public class callCDICurrent implements Runnable {
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.api_fat/test-applications/cdiCurrentTest.war/src/com/ibm/ws/cdi12/test/common/web/CDICurrentTestServlet.java:97]: I thought we already had a test for a vanilla CDI.current() call?

**Comment by benjamin-confino** [comment]
#build

**Review by tevans78 — APPROVED** [review]

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Pull request merged**